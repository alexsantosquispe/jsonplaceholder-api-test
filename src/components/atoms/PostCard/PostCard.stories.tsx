import type { Meta, StoryObj } from '@storybook/react';

import { PostCard } from './PostCard';
import { Wrapper } from '../../../utils/stories-utils';

const meta = {
  title: 'Atoms/PostCard',
  component: PostCard,
  decorators: [
    (Story) => (
      <div style={{ margin: '1rem', width: '300px' }}>
        <Story />
      </div>
    )
  ]
} satisfies Meta<typeof PostCard>;

export default meta;

type Story = StoryObj<typeof PostCard>;

export const Default: Story = () => {
  return (
    <Wrapper>
      <PostCard
        id={1}
        title="Lorem ipsum dolor sit amet"
        body="Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quos."
      />
    </Wrapper>
  );
};

Default.args = {};
