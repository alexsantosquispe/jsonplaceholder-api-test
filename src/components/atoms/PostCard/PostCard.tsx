import { useState } from 'react';
import { twMerge } from 'tailwind-merge';

import { Post } from '../../../types';
import { PostDetailModal } from '../PostDetailModal/PostDetailModal';

type PostCardProps = Omit<Post, 'userId'>;

export const PostCard = ({ id, title, body }: PostCardProps) => {
  const [isOpen, setIsOpen] = useState(false);

  const onClose = () => {
    document.body.style.overflow = 'unset';
    setIsOpen(false);
  };

  const openModal = () => {
    document.body.style.overflow = 'hidden';
    setIsOpen(true);
  };

  return (
    <>
      <a
        key={id}
        className={twMerge(
          'flex h-auto w-full flex-col gap-4 rounded-lg border px-4 py-6',
          'border-gray-200 text-black hover:cursor-pointer hover:bg-gray-100',
          'dark:border-white/10 dark:text-white dark:hover:border-white/10 dark:hover:bg-white/10'
        )}
        onClick={openModal}
      >
        <h4 className="text-base leading-7 font-semibold capitalize">
          {title}
        </h4>
        <p className="text-sm leading-6 font-normal normal-case">{body}.</p>
      </a>
      {isOpen && <PostDetailModal postId={String(id)} onClose={onClose} />}
    </>
  );
};
