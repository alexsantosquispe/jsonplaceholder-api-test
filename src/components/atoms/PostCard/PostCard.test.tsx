import { render, screen } from '@testing-library/react';

import { PostCard } from './PostCard';

describe('PostCard', () => {
  const props = {
    id: 1,
    title: 'post card test',
    body: 'post card test body'
  };

  it('should match snapshot', () => {
    const component = render(<PostCard {...props} />);

    expect(component).toMatchSnapshot();

    expect(screen.getByText(props.title)).toBeInTheDocument();
  });
});
