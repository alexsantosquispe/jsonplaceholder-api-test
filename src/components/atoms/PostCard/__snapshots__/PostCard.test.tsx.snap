// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard should match snapshot 1`] = `
<div>
  <div
    class="flex h-auto w-full flex-col gap-2 rounded-lg border p-3 border-gray-200 hover:cursor-pointer hover:bg-gray-100 dark:border-white/10 dark:hover:bg-white/10"
  >
    <span
      class="text-base leading-7 font-medium capitalize md:text-sm md:leading-6"
    >
      post card test
    </span>
    <p
      class="text-primary/70 text-sm leading-6 font-normal normal-case md:text-xs md:leading-5 dark:text-white/70"
    >
      post card test body
      .
    </p>
  </div>
</div>
`;
